grakn-core/src/main/java/ai/grakn/concept/LabelId.java

```diff
@@ -34,49 +35,24 @@
  *
  * @author fppt
  */
-public class LabelId implements Comparable<LabelId>, Serializable {
-    private static final long serialVersionUID = 3181633335040468179L;
-
-    private Integer typeId;
-
-    private LabelId(Integer typeId){
-        Preconditions.checkNotNull(typeId);
-        this.typeId = typeId;
-    }
+@AutoValue
+public abstract class LabelId implements Comparable<LabelId>, Serializable {
+    private static final long serialVersionUID = -1676610785035926909L;
 
     /**
      *
      * @return Used for indexing purposes and for graql traversals
      */
-    public Integer getValue(){
-        return typeId;
-    }
-
-    @Override
-    public boolean equals(Object o) {
```

Comment:
On that note, How does autovalue handle equality and the hashcode? Is it based on the fields somehow? because if so I imagine a little room for error there.